hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1000,70 +1001,18 @@ def shrink(self, example, predicate):
     def new_shrinker(self, example, predicate):
         return Shrinker(self, example, predicate)
 
-    def prescreen_buffer(self, buffer):
-        """Attempt to rule out buffer as a possible interesting candidate.
-
-        Returns False if we know for sure that running this buffer will not
-        produce an interesting result. Returns True if it might (because it
-        explores territory we have not previously tried).
-
-        This is purely an optimisation to try to reduce the number of tests we
-        run. "return True" would be a valid but inefficient implementation.
-        """
-
-        # Traverse the tree, to see if we have already tried this buffer
-        # (or a prefix of it).
-        node_index = 0
-        n = len(buffer)
-        for k, b in enumerate(buffer):
-            if node_index in self.dead:
-                # This buffer (or a prefix of it) has already been tested,
-                # or has already had its descendants fully explored.
-                # Testing it again would not be helpful.
-                return False
-            try:
-                # The block size at that point provides a lower bound on how
-                # many more bytes are required. If the buffer does not have
-                # enough bytes to fulfill that block size then we can rule out
-                # this buffer.
-                if k + self.block_sizes[node_index] > n:
-                    return False
-            except KeyError:
-                pass
-
-            # If there's a forced value or a mask at this position, then
-            # pretend that the buffer already contains a matching value,
-            # because the test function is going to do the same.
-            try:
-                b = self.forced[node_index]
-            except KeyError:
-                pass
-            try:
-                b = b & self.masks[node_index]
-            except KeyError:
-                pass
-
-            try:
-                node_index = self.tree[node_index][b]
-            except KeyError:
-                # The buffer wasn't in the tree, which means we haven't tried
-                # it. That makes it a possible candidate.
-                return True
-        else:
-            # We ran out of buffer before reaching a leaf or a missing node.
-            # That means the test function is going to draw beyond the end
-            # of this buffer, which makes it a bad candidate.
-            return False
-
     def cached_test_function(self, buffer):
         """Checks the tree to see if we've tested this buffer, and returns the
         previous result if we have.
 
         Otherwise we call through to ``test_function``, and return a
         fresh result.
         """
+        rewritten = bytearray()
+        overrun = False
```

Comment:
Minor: Perhaps `would_overrun` would be clearer here, since it's a (reliable) prediction rather than an after-the-fact observation.